src/analysis/processing/qgsalgorithmextractlabels.cpp

```diff
@@ -0,0 +1,372 @@
+
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::MultiLineWrapChar ) )
+      {
+        labelSettings.wrapChar = dataDefinedValues.value( QgsPalLayerSettings::MultiLineWrapChar ).toString();
+      }
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::AutoWrapLength ) )
+      {
+        labelSettings.autoWrapLength = dataDefinedValues.value( QgsPalLayerSettings::AutoWrapLength ).toInt();
+      }
+      const QString labelText = QgsPalLabeling::splitToLines( labelFeature->text( -1 ),
+                                labelSettings.wrapChar,
+                                labelSettings.autoWrapLength,
+                                labelSettings.useMaxLineLengthForAutoWrap ).join( "\n" );
+
+      QString labelAlignment;
+      if ( dataDefinedValues.contains( QgsPalLayerSettings::MultiLineAlignment ) )
+      {
+        labelSettings.multilineAlign = static_cast< QgsPalLayerSettings::MultiLineAlign >( dataDefinedValues.value( QgsPalLayerSettings::MultiLineAlignment ).toInt() );
+      }
+      switch ( labelSettings.multilineAlign )
+      {
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiRight:
+          labelAlignment = QStringLiteral( "right" );
+          break;
+
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiCenter:
+          labelAlignment = QStringLiteral( "center" );
+          break;
+
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiLeft:
+          labelAlignment = QStringLiteral( "left" );
+          break;
+
+        case QgsPalLayerSettings::QgsPalLayerSettings::MultiJustify:
+          labelAlignment = QStringLiteral( "justify" );
+          break;
+
+        case QgsPalLayerSettings::MultiFollowPlacement:
+          switch ( label->getQuadrant() )
+          {
+            case pal::LabelPosition::QuadrantAboveLeft:
+            case pal::LabelPosition::QuadrantLeft:
+            case pal::LabelPosition::QuadrantBelowLeft:
+              labelAlignment = QStringLiteral( "right" );
+              break;
+
+            case pal::LabelPosition::QuadrantAbove:
+            case pal::LabelPosition::QuadrantOver:
+            case pal::LabelPosition::QuadrantBelow:
+              labelAlignment = QStringLiteral( "center" );
+              break;
+
+            case pal::LabelPosition::QuadrantAboveRight:
+            case pal::LabelPosition::QuadrantRight:
+            case pal::LabelPosition::QuadrantBelowRight:
+              labelAlignment = QStringLiteral( "left" );
+              break;
+          }
+          break;
+      }
+
+      const double labelRotation = label->getAlpha() * 180 / M_PI;
+
+      const QFont font = labelFeature->definedFont();
+      const QString fontFamily = font.family();
+      const QString fontStyle = font.styleName();
+      const double fontSize = static_cast<double>( font.pixelSize() ) * 72 / context.painter()->device()->logicalDpiX();
```

Comment:
where's the 72 from?

Reply:
@nyalldawson , 72 was what I've been using to do pixel to point conversion. I found that the labelFeature->definedFont() didn't have a proper pointSize() value (returns -1), and I don't think we should store pixel size value as it doesn't take scale in consideration.

Reply:
Why 72 though? Shouldn't this be at least 96? (We really should store the text format in QgsTextLabelFeature instead!)